workers/unity/Packages/io.improbable.gdk.core/Representation/Editor/EntityRepresentationMappingEditor.cs

```diff
@@ -0,0 +1,116 @@
+using System;
+using UnityEditor;
+using UnityEditor.UIElements;
+using UnityEngine;
+using UnityEngine.UIElements;
+
+namespace Improbable.Gdk.Core.Representation.Editor
+{
+    [CustomEditor(typeof(EntityRepresentationMapping))]
+    public class EntityRepresentationMappingEditor : UnityEditor.Editor
+    {
+        private static TypeCache.TypeCollection entityRepresentationTypes;
+
+        private EntityRepresentationMapping TargetDatabase => (EntityRepresentationMapping) target;
+        private SerializedProperty listProperty;
+
+        private VisualElement listContainer;
+
+        static EntityRepresentationMappingEditor()
+        {
+            entityRepresentationTypes = TypeCache.GetTypesDerivedFrom<IEntityRepresentationResolver>();
+        }
+
+        private void OnEnable()
+        {
+            listProperty = serializedObject.FindProperty(nameof(EntityRepresentationMapping.EntityRepresentationResolvers));
+        }
+
+        public override VisualElement CreateInspectorGUI()
+        {
+            var container = new VisualElement { name = "container" };
+            container.style.paddingTop = 5;
+
+            listContainer = new VisualElement { name = "list-container" };
+            container.Add(listContainer);
+            GenerateListElements();
+
+            var addButton = CreateNewEntityButton();
+            container.Add(addButton);
+
+            return container;
+        }
+
+        private void GenerateListElements()
+        {
+            listContainer.Clear();
+            for (var i = 0; i < listProperty.arraySize; i++)
+            {
+                var elementIndex = i;
+                var property = listProperty.GetArrayElementAtIndex(i);
+
+                var superElementContainer = new VisualElement { name = "element-root-container" };
+                var typeLabel = new Label(TargetDatabase.EntityRepresentationResolvers[i].GetType().Name)
+                {
+                    name = "element-type-name"
```

Comment:
Debugging, I can remove them if you rather dont have them

Reply:
Yeah I'd rather remove them unless they are being functionally used. Especially if you move styles to USS, you'll have classes on each of these instead, it will be a bit redundant